src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_newBooking.kt

```diff
@@ -1199,16 +1199,29 @@ class OffenderResourceIntTest_newBooking : ResourceTest() {
     }
   }
 
-  fun createActiveBooking(prisonId: String = "MDI"): String = OffenderBuilder().withBooking(
-    OffenderBookingBuilder(
-      prisonId = prisonId,
-    ),
-  ).save(testDataContext).offenderNo
-
-  fun createInactiveBooking(iepLevel: String = "ENH"): String = OffenderBuilder().withBooking(
-    OffenderBookingBuilder(
-      prisonId = "MDI",
-      released = true,
-    ).withIEPLevel(iepLevel),
-  ).save(testDataContext).offenderNo
+  fun createActiveBooking(prisonId: String = "MDI"): String {
+    lateinit var offenderNo: String
+
+    builder.build {
+      offenderNo = offender {
+        booking(prisonId = prisonId)
+      }.offenderNo
+    }
+
+    return offenderNo
+  }
+
+  fun createInactiveBooking(): String {
+    lateinit var offenderNo: String
+
+    builder.build {
+      offenderNo = offender {
+        booking(prisonId = "MDI") {
+          release()
+        }
+      }.offenderNo
+    }
+
+    return offenderNo
+  }
```

Comment:
Are there any tests here where there is no booking at all - the new booking is the first one?

Reply:
Yes in the NewOffender nested test